### PR TITLE
Migrated oauth flow to db for google calendar

### PR DESCRIPTION
**Pull request by Tarraann**
### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [x] I have read the contributing guide and my code conforms to the guidelines.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have added the required tests.

### CODE REVIEW

File: superagi/tools/google_calendar/delete_calendar_event.py [hunk 1]
```diff
@@ -12,8 +14,11 @@ class DeleteCalendarEventTool(BaseTool):
     description: str = "Delete an event from Google Calendar"
 
     def _execute(self, event_id: str):
+        engine = connect_db()
```
Comment (TransformerOptimus) [inline superagi/tools/google_calendar/delete_calendar_event.py:17]: Session is already there in self

File: superagi/controllers/google_oauth.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        'code': code,
+        'access_type': 'offline'
+    }
+    response = requests.post(token_uri, data=params)
+    response = response.json()
+    expire_time = datetime.utcnow() + timedelta(seconds=response['expires_in'])
+    expire_time = expire_time - timedelta(minutes=5)
+    response['expiry'] = expire_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+    response_data = json.dumps(response)
+    print("/////////////////////")
```
Comment (TransformerOptimus) [inline superagi/controllers/google_oauth.py:44]: Remove prints

File: superagi/tools/google_calendar/create_calendar_event.py [hunk 3]
```diff
@@ -20,8 +22,11 @@ class CreateEventCalendarTool(BaseTool):
     description: str = "Create an event for Google Calendar"
 
     def _execute(self, event_name: str, description: str, attendees: list, start_date: str = 'None', start_time: str = 'None', end_date: str = 'None', end_time: str = 'None', location: str = 'None'):
+        engine = connect_db()
+        Session = sessionmaker(bind=engine)
+        session = Session()
         toolkit_id = self.toolkit_config.toolkit_id
-        service = GoogleCalendarCreds().get_credentials(toolkit_id)
+        service = GoogleCalendarCreds(session).get_credentials(toolkit_id)
```
Comment (TransformerOptimus) [inline superagi/tools/google_calendar/create_calendar_event.py:29]: use the tools base session

File: superagi/tools/google_calendar/delete_calendar_event.py [hunk 4]
```diff
@@ -12,8 +14,9 @@ class DeleteCalendarEventTool(BaseTool):
     description: str = "Delete an event from Google Calendar"
 
     def _execute(self, event_id: str):
+        session = self.toolkit_config.session
         toolkit_id = self.toolkit_config.toolkit_id
-        service = GoogleCalendarCreds().get_credentials(toolkit_id)
+        service = GoogleCalendarCreds(session).get_credentials(toolkit_id)
```
Comment (TransformerOptimus) [inline superagi/tools/google_calendar/delete_calendar_event.py:19]: u can directly use it no need for temp var.

File: superagi/tools/google_calendar/event_details_calendar.py [hunk 5]
```diff
@@ -13,8 +15,11 @@ class EventDetailsCalendarTool(BaseTool):
     description: str = "Fetch an event from Google Calendar"
 
     def _execute(self, event_id: str):
+        engine = connect_db()
+        Session = sessionmaker(bind=engine)
+        session = Session()
         toolkit_id = self.toolkit_config.toolkit_id
-        service = GoogleCalendarCreds().get_credentials(toolkit_id)
+        service = GoogleCalendarCreds(session).get_credentials(toolkit_id)
```
Comment (TransformerOptimus) [inline superagi/tools/google_calendar/event_details_calendar.py:22]: use base session here.

**Review by Fluder-Paradyne — APPROVED** [review]

**Pull request merged**